Bug: Type error when using ConfigWithExtends

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Playground Link

_No response_

### Repro Code

I get a TypeScript type error `scripts-eslint/eslint.config.mjs:7:2 - error TS2353: Object literal may only specify known properties, and 'ignores' does not exist in type 'ConfigWithExtends'` when using the `ConfigWithExtends` type in the following example. 

```TypeScript
// @ts-check

import tseslint from 'typescript-eslint';

/** @type {import('typescript-eslint').ConfigWithExtends} */
const config_base = {
	ignores: [
		'temp/**/*',
	],
};

/** @type {import('typescript-eslint').ConfigWithExtends[]} */
const configs = [config_base];

const flat_config = tseslint.config(...configs);

export {flat_config};
```


### ESLint Config

_No response_

### tsconfig

_No response_

### Expected Result

n.a.

### Actual Result

n.a.

### Additional Info

[flatconfig_error_1.zip](https://github.com/typescript-eslint/typescript-eslint/files/14655262/flatconfig_error_1.zip)


### Versions

<!-- Fill in the relevant versions and delete the others -->
| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint`             | `7.3.1` |
| `TypeScript`                           | `5.4.3` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `20.11.0` |

> ### Before You File a Bug Report Please Confirm You Have Done The Following...
> 在提交错误报告之前，请确认您已完成以下操作...
> * [x]  I have tried restarting my IDE and the issue persists. 我尝试重新启动 IDE，但问题仍然存在。
> * [x]  I have updated to the latest version of the packages. 我已经更新到最新版本的软件包。
> * [x]  I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue. 我搜索了相关问题，但没有发现与我的问题相符的问题。
> * [x]  I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed. 我已阅读常见问题解答，但我的问题未列出。
> 
> ### Relevant Package 相关套餐
> typescript-eslint
> 
> ### Playground Link 游乐场链接
> _No response 没有反应_
> 
> ### Repro Code 复制代码
> ```ts
> I get a TypeScript type error `scripts-eslint/eslint.config.mjs:7:2 - error TS2353: Object literal may only specify known properties, and 'ignores' does not exist in type 'ConfigWithExtends'` when using the `ConfigWithExtends` type in the following example.
> 
> 
> // @ts-check
> 
> import tseslint from 'typescript-eslint';
> 
> /** @type {import('typescript-eslint').ConfigWithExtends} */
> const config_base = {
> 	ignores: [
> 		'temp/**/*',
> 	],
> };
> 
> /** @type {import('typescript-eslint').ConfigWithExtends[]} */
> const configs = [config_base];
> 
> const flat_config = tseslint.config(...configs);
> 
> export {flat_config};
> ```
> 
> ```
> 
> 
> ### ESLint Config
> 
> _No response_
> 
> ### tsconfig
> 
> _No response_
> 
> ### Expected Result
> 
> n.a.
> 
> ### Actual Result
> 
> n.a.
> 
> ### Additional Info
> 
> [flatconfig_error_1.zip](https://github.com/typescript-eslint/typescript-eslint/files/14655262/flatconfig_error_1.zip)
> 
> 
> ### Versions
> 
> <!-- Fill in the relevant versions and delete the others -->
> | package                                | version |
> | -------------------------------------- | ------- |
> | `typescript-eslint`             | `7.3.1` |
> | `TypeScript`                           | `5.4.3` |
> | `ESLint`                               | `8.57.0` |
> | `node`                                 | `20.11.0` |
> ```

Now the typescript can only see the `extends` field in `ConfigWithExtends` interface，this explains why other field of `Config` including `ignores` will be treated as no exist.

--- a/packages/typescript-eslint/package.json
+++ b/packages/typescript-eslint/package.json
@@ -51,12 +51,12 @@
     "typecheck": "tsc -p tsconfig.json --noEmit"
   },
   "peerDependencies": {
-    "@typescript-eslint/utils": "7.3.1",
     "eslint": "^8.56.0"
   },
   "dependencies": {
     "@typescript-eslint/eslint-plugin": "7.3.1",
-    "@typescript-eslint/parser": "7.3.1"
+    "@typescript-eslint/parser": "7.3.1",
+    "@typescript-eslint/utils": "7.3.1"
   },
   "devDependencies": {
     "downlevel-dts": "*",
@@ -66,9 +66,6 @@
     "typescript": "*"
   },
   "peerDependenciesMeta": {
-    "@typescript-eslint/utils": {
-      "optional": true
-    },
     "typescript": {
       "optional": true
     }

--- a/packages/typescript-eslint/package.json
+++ b/packages/typescript-eslint/package.json
@@ -51,6 +51,7 @@
     "typecheck": "tsc -p tsconfig.json --noEmit"
   },
   "peerDependencies": {
+    "@typescript-eslint/utils": "7.3.1",
     "eslint": "^8.56.0"
   },
   "dependencies": {
@@ -65,6 +66,9 @@
     "typescript": "*"
   },
   "peerDependenciesMeta": {
+    "@typescript-eslint/utils": {
+      "optional": true
+    },
     "typescript": {
       "optional": true
     }

